cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceHandler.java

```diff
@@ -100,52 +103,50 @@ public DatasetInstanceHandler(DatasetTypeManager implManager, DatasetInstanceMan
-      datasetSummaries.add(new DatasetSpecificationSummary(spec.getName(), spec.getType(), spec.getProperties()));
-    }
-    responder.sendJson(HttpResponseStatus.OK, datasetSummaries);
+    Collection<DatasetSpecification> specs = instanceManager.getAll(Id.Namespace.from(namespaceId));
+    responder.sendJson(HttpResponseStatus.OK, spec2Summary(specs));
   }
 
   @GET
   @Path("/data/datasets/{name}")
-  public void getInfo(HttpRequest request, HttpResponder responder,
-                      @PathParam("namespace-id") String namespaceId,
-                      @PathParam("name") String name,
-                      @QueryParam("owner") List<String> owners) {
+  public void get(HttpRequest request, HttpResponder responder,
+                  @PathParam("namespace-id") String namespaceId,
+                  @PathParam("name") String name,
+                  @QueryParam("owner") List<String> owners) throws NotFoundException {
 
     Id.DatasetInstance datasetId = Id.DatasetInstance.from(namespaceId, name);
     DatasetSpecification spec = instanceManager.get(datasetId);
     if (spec == null) {
-      responder.sendStatus(HttpResponseStatus.NOT_FOUND);
-    } else {
-      // try finding type info in the dataset's namespace first, then the system namespace
-      DatasetTypeMeta typeMeta = getTypeInfo(Id.Namespace.from(namespaceId), spec.getType());
-      if (typeMeta == null) {
-        // Dataset type not found in the instance's namespace or the system namespace. Bail out.
-        responder.sendString(HttpResponseStatus.NOT_FOUND,
-                             String.format("Dataset type %s used by dataset %s not found", spec.getType(), name));
-        return;
-      }
-      // typeMeta is guaranteed to be non-null now.
-      DatasetMeta info = new DatasetMeta(spec, typeMeta, null);
-      for (String owner : owners) {
-        String[] parts = owner.split("::", 2);
-        Preconditions.checkArgument(parts.length == 2);
-        String ownerType = parts[0];
-        String ownerId = parts[1];
-        try {
-          if (ownerType.equals(Id.getType(Id.Program.class))) {
-            usageRegistry.register(Id.Program.fromStrings(ownerId.split("/")), datasetId);
-          } else if (ownerType.equals(Id.getType(Id.Adapter.class))) {
-            usageRegistry.register(Id.Adapter.fromStrings(ownerId.split("/")), datasetId);
-          }
-        } catch (Exception e) {
-          LOG.warn("Failed to register usage of {} -> {}", ownerId, datasetId);
+      throw new NotFoundException(datasetId);
+    }
+
+    Id.DatasetType datasetTypeId = Id.DatasetType.from(namespaceId, spec.getType());
+    DatasetTypeMeta typeMeta = getTypeInfo(Id.Namespace.from(namespaceId), spec.getType());
+    if (typeMeta == null) {
+      // TODO: This shouldn't happen unless CDAP is in an invalid state - maybe give different error
+      throw new NotFoundException(datasetTypeId);
+    }
+
+    // register ownership
+    for (String owner : owners) {
```

Comment:
Not good to have logic in handler (I know there are existing one, but we all know we want to migrate away from the current way).